pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java

```diff
@@ -267,6 +267,48 @@ public void revokePermissionOnSubscription(@PathParam("property") String propert
         internalRevokePermissionsOnSubscription(subscription, role);
     }
 
+    @PUT
+    @Path("/{property}/{namespace}/implicitPermissionOnSubscription")
+    @ApiOperation(hidden = true, value = "Allow a consumer's role to have implicit permission to consume from a"
+            + " subscription.")
+    @ApiResponses(value = {@ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Property or cluster or namespace doesn't exist"),
+            @ApiResponse(code = 409, message = "Concurrent modification"),
+            @ApiResponse(code = 501, message = "Authorization is not enabled")})
+    public void grantImplicitPermissionOnSubscription(
+            @PathParam("property") String property,
+            @PathParam("namespace") String namespace) {
+        validateNamespaceName(property, namespace);
+        internalSetImplicitPermissionOnSubscription(true);
```

Comment:
It might be worth considering PIP-149/#14365 here and use the async style to implement the API.

To clarify it: the server side implementation should by async, but the client API can contain both sync and async API methods.

Reply:
Great point. I'll update the server's code.

Reply:
I've implemented the server using completely async methods. Please take a look.